src/Compilers/Core/MSBuildTask/ManagedCompiler.cs

```diff
@@ -501,15 +502,16 @@ protected override int ExecuteTool(string pathToTool, string responseFileCommand
 
             try
             {
-                using var logger = new CompilerServerLogger();
+                using var logger = new CompilerServerLogger($"MSBuild {Process.GetCurrentProcess().Id}");
```

Comment:
Do we really need to include the process ID into the identifier string, since `CompilerServerLogger` already captures the process ID?

Reply:
The `CompilerServerLogger` only uses the process ID when generating the name for the log file on disk, it doesn't actually use that ID within the log. The use of the process ID here is important because there are multiple MSBuild process instances in the build and we want to differentiate between them. Unlike VBSCompiler of which there is one hence not a lot of value in dumping that in the log.